linenum.c

```diff
@@ -199,14 +199,17 @@ add_lnum(LINENUM linenum, off_t pos)
 }
 
 static int loopcount;
-static time_t startime;
+static struct timespec timeout;
 
 static void
 longish(void)
 {
+	struct timespec now;
+
 	if (loopcount >= 0 && ++loopcount > 100) {
 		loopcount = 0;
-		if (time(NULL) >= startime + LONGTIME) {
+		clock_gettime(CLOCK_MONOTONIC, &now);
```

Comment:
See my review feedback in the conversation -- basically I think we need to use this new logic only if CLOCK_MONOTONIC is defined -- on SUSv2 systems it isn't.